src/features/price/price.services.ts

```diff
@@ -0,0 +1,49 @@
+import BigNumber from 'bignumber.js';
+import config from '@/configuration';
+import { getRequest, gqlRequest } from '@/helpers/requests';
+import {
+	FETCH_MAINNET_TOKEN_PRICE,
+	FETCH_GNOSIS_TOKEN_PRICE,
+} from './price.queries';
+
+export const fetchEthPrice = async (): Promise<string> => {
+	const res = await getRequest(
+		'https://feathers.giveth.io/conversionRates?from=ETH&to=USD&interval=hourly',
+	);
+	return res.rate;
+};
+
+export const fetchMainnetTokenPrice = async (
+	tokenId: string,
+): Promise<string> => {
+	const query = FETCH_MAINNET_TOKEN_PRICE;
+	const variables = {
+		tokenId: tokenId.toLowerCase(),
+		daiId: '0x6b175474e89094c44da98b954eedeac495271d0f'.toLowerCase(),
+	};
+	const { data } = await gqlRequest(
+		config.MAINNET_CONFIG.uniswapV2Subgraph,
+		false,
+		query,
+		variables,
+	);
+	const givEth = new BigNumber(data.givtoken.derivedETH);
```

Comment:
Based on the query in `src/features/price/price.queries.ts`, it should be
```suggestion
	const tokenEthPrice = new BigNumber(data.token.derivedETH);
```

Also, as the aimed token is not always `GIV`, we can rename the variable name.